### PR TITLE
Add ActivityLogBanner and specific versions to show restore status

### LINKED ISSUE #14664: Activity Log/Rewind: Restoring site (progress, fail, and success) component.
The restore process is a copy/paste from @jeffgolenski‘s great work: Activity Log: Restore flows for v.1. A progress bar (or just a banner with a message) shows you how far along in the process the site restore is.

image

If it fails, we should try to fire it again automatically. But in case that also fails, we should let the user know.

image

If it succeeds, we show a confirmation. 

image

### PR DESCRIPTION
**Pull request by sirreal**
These banners share some structure, but their props and some internal structure are different. I've tried to find a good generalized component and specialize it for our 3 use cases.

Add `ActivityLogBanner` (generalized component).
Add `ActivityLogErrorBanner`.
Add `ActivityLogSuccessBanner`.
Add `ActivityLogProgressBanner`.

Notable variations from the design (ping @jeffgolenski @keoshi):
* I used `is-success` styling (green) for the success banner. We can switch back to blue, just let me know.
* I used a suggestion form @a8ci18n for one of the button texts, becomes "Thanks, got it!".
* Used moment's native `LLLL` format for long dates.

activity-log-banners

Todo:
- ~Define what props will be passed to each specific component (error, progress, success).~
- ~Implement props logic~
- ~Implement banner selection in `ActivityLog`~
- [x] Simplify highlighting (the left border) with #14564 when it's merged

Fixes #14664
Replaces #14646

### CODE REVIEW

File: client/my-sites/stats/activity-log/activity-log-success-banner.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+			<p>{ translate(
+				'We successfully restored your site back to %s!',
+				{ args: moment( date ).format( 'LLLL' ) }
+			) }</p>
+			<Button primary >
+				{ translate( 'View site' ) }
+			</Button>
+			{ '  ' }
+			<Button onClick={ handleDismiss }>
+				{ translate( 'Got it, thanks' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log/activity-log-success-banner.jsx]: Hi! I've found a possible matching string that has already been translated **10** times:
`translate( 'Thanks, got it!' )` *ES Score: 13*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/stats/activity-log/activity-log-success-banner.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	const date = 1496400468285;
+	// FIXME: real dismiss
+	const handleDismiss = () => {};
+
+	return (
+		<ActivityLogBanner
+			isDismissable
+			onDismissClick={ handleDismiss }
+			status="is-success"
+			title={ translate( 'Your site has been successfully restored' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log/activity-log-success-banner.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'has been successfully restored!' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/stats/activity-log-banner/activity-log-success-banner.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	const date = 1496400468285;
+	// FIXME: real dismiss
+	const handleDismiss = () => {};
+
+	return (
+		<ActivityLogBanner
+			isDismissable
+			onDismissClick={ handleDismiss }
+			status="is-success"
+			title={ translate( 'Your site has been successfully restored' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-banner/activity-log-success-banner.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'has been successfully restored!' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by dereksmart** [comment]
Couple of thoughts: 
- are the file names `/activity-log-banner/activity-log-error-banner.jsx` redundant?  It could be `/activity-log-banner/error-banner.jsx`. `/activity-log-banner/progress-banner.jsx`, etc
- The success and error components are structurally identical, just different status, text, and icon.  Does it make sense to combine there?

**Comment by keoshi** [comment]
Should we change

> “You'll receive a notification once it's complete.”

to

> “You'll be notified once it's complete.”

?

It abstracts it a little bit — we'll also be sending an email, mobile notifications, and updating the UI. @jeffgolenski

**Comment by rickybanister** [comment]
I like that change—it allows us to adhere to their respective notification settings. 

It also looks like the icon is unevenly spaced—it's a little closer to the edge of the card than to the text.

**Comment by rickybanister** [comment]
Also @tyxla recently introduced the 'highlight' prop for `card`. Can/should we use that here rather than duplicate some of that same functionality in this new component? We obviously still need our own component, but perhaps we can inherit the border style from card itself?

**Comment by tyxla** [comment]
Just to clarify: #14564 is the PR that @rickybanister is talking about, and it was merged just a minute ago.

File: client/my-sites/stats/activity-log-banner/activity-log-success-banner.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	const date = 1496400468285;
+	// FIXME: real dismiss
+	const handleDismiss = () => {};
+
+	return (
+		<ActivityLogBanner
+			isDismissable
+			onDismissClick={ handleDismiss }
+			status="success"
+			title={ translate( 'Your site has been successfully restored' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-banner/activity-log-success-banner.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'has been successfully restored!' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by sirreal** [comment]
I've addressed the design issues mentioned by @rickybanister and @keoshi:
* Text change
* Alignment
* Use Card highlighting (@tyxla)

**Comment by sirreal** [comment]
@dereksmart 
> are the file names `/activity-log-banner/activity-log-error-banner.jsx` redundant?

Good point, I reduced some verbosity in e3f126d. I'm hesitant to rename the generic `ActivityLogBanner` because we already have a `Banner` component.

> The success and error components are structurally identical, just different status, text, and icon. Does it make sense to combine there?

I feel like this is a pretty good balance between the generic and specific components. There absolutely could be an abstraction over just those two versions of the banner, but I don't know that it will offer much benefit.

File: client/my-sites/stats/activity-log-banner/activity-log-banner.jsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+			onDismissClick,
+			children,
+			title,
+			translate,
+			status,
+		} = this.props;
+
+		const icon = this.getIcon();
+		const classes = classNames( 'activity-log-banner', {
+			'is-error': 'error' === status,
```
Comment (tyxla) [inline client/my-sites/stats/activity-log-banner/activity-log-banner.jsx]: You don't need to specify any of these classes - if you use `highlight`, these classes will be added automatically to the `Card`.

File: client/my-sites/stats/activity-log-banner/success-banner.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	const date = 1496400468285;
+	// FIXME: real dismiss
+	const handleDismiss = () => {};
+
+	return (
+		<ActivityLogBanner
+			isDismissable
+			onDismissClick={ handleDismiss }
+			status="success"
+			title={ translate( 'Your site has been successfully restored' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log-banner/success-banner.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( 'has been successfully restored!' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by sirreal** [comment]
I've updated the to-do, I think this component is ready to merge and the props/state changes are tagged with `FIXME`s and can be cleaned up when other parts are in place.

**Comment by dereksmart** [comment]
LGTM! 🐑

File: client/my-sites/stats/activity-log-banner/activity-log-banner.jsx (hunk 5 above)
Comment (sirreal) [inline client/my-sites/stats/activity-log-banner/activity-log-banner.jsx]: Awesome! Thanks for the heads up, removed a bunch of now redundant code in 5a4e5fe

**Pull request merged**